compiler/rustc_hir_typeck/src/callee.rs

```diff
@@ -502,18 +502,12 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
         let fn_sig = self.instantiate_binder_with_fresh_vars(call_expr.span, infer::FnCall, fn_sig);
         let fn_sig = self.normalize(call_expr.span, fn_sig);
 
-        // Call the generic checker.
-        let expected_arg_tys = self.expected_inputs_for_expected_output(
-            call_expr.span,
-            expected,
-            fn_sig.output(),
-            fn_sig.inputs(),
-        );
         self.check_argument_types(
```

Comment:
why not just pass in the whole `fn_sig` :thinking: we currently always pass in inputs, outputs, and c_variadic manually

Reply:
ah, `check_argument_types` is also used during diagnostics 🤔 that code should also be able to use a full error dumm `FnSig` though 🤔